druntime/src/core/atomic.d

```diff
@@ -1536,6 +1536,18 @@ struct Atomic(T)
         return m_val.atomicLoad!(order)();
     }
 
+    /**
+    * Atomically loads and returns the current value of the atomic variable.
+    * Equivalent to load().
+    *
+    * Returns:
+    *  The atomic value.
+    */
+    T opImplicitCast(T)() pure nothrow @nogc
```

Comment:
this is too general, since you're already inside Atomic(T) this should specialize specifically to the original T

T opImplicitCast(R : T)() {}

something like that. As it is written right now it would try to trigger for string and stuff too.